CMakeLists.txt

```diff
@@ -1284,6 +1286,37 @@ else()
   list(APPEND Sources ${GUI_SOURCES} ${GUI_HEADER_ONLY} ${GUI_UIS} ${GUI_HEADERS})
 endif()
 
+if(ENABLE_GUI_TESTS)
+  set(GUITESTS_HEADERS
+    src/guitests/guitests.h
+    src/guitests/TestModuleCache.h
+    src/guitests/TestTabManager.h
+    src/guitests/TestMainWindow.h
+    src/guitests/UXTest.h
+  )
+
+  set(GUITESTS_SOURCES
+    src/guitests/guitests.cc
+    src/guitests/TestModuleCache.cc
+    src/guitests/TestTabManager.cc
+    src/guitests/TestMainWindow.cc
+    src/guitests/UXTest.cc
+  )
+
+  list(APPEND Sources ${GUITESTS_SOURCES} ${GUITESTS_HEADERS})
+  target_compile_definitions(OpenSCAD PRIVATE HAS_GUI_TESTS=true)
+else()
+  set(GUITESTS_HEADERS
+    src/guitests/guitests.h
+  )
+
+  set(GUITESTS_SOURCES
+    src/guitests/guitests.cc
+  )
+  list(APPEND Sources ${GUITESTS_SOURCES} ${GUITESTS_HEADERS})
+  target_compile_definitions(OpenSCAD PRIVATE HAS_GUI_TESTS=false)
```

Comment:
I think it will be cleaner to just avoid defining HAS_GUI_TESTS and not add the guitests.* files, and checking the  HAS_GUI_TESTS macro at the call site.